nebula-unity/Assets/scripts/DungeonMapGenerator.cs

```diff
@@ -108,11 +109,15 @@ public class PotentialDoor {
     }
 
     // This function will generate a map given some bounding dimensions
-    public Map GenerateMap(int xSize = XDIMDEFAULT, int ySize = YDIMDEFAULT) {
+    public Map GenerateMap(int xSize, int ySize, int minDiam, int maxDiam, float sfact, int maxatt) {
         ResetGenerationVariables();
 
         mapSize.x = xSize;
         mapSize.y = ySize;
+        minRoomDiameter = minDiam;
+        maxRoomDiameter = maxDiam;
+        shapeFactor = sfact;
+        maxAttempts = maxatt;
```

Comment:
Since you removed the only reference to them, you should also remove XDIMDEFAULT and YDIMDEFAULT.

....or a better idea is to use defaults for all of those variables!   If we can set those values in the inspector, we should use them as defaults!   Then we can just call MakeMeAMap() without params to keep remaking maps with the same params!

Reply:
If we do it this way, lines 115-120 become redundant and we wont need them anymore either.  Assuming we have a function declaration something like this:

`public Map GenerateMap(int xSize = mapSize.x, int ySize = maxSize.y, int minDiam = minRoomDiameter, int maxDiam = maxRoomDiameter, float sfact = shapeFactor, int maxatt = maxAttempts)`

We can then use [keyword arguments](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments) to call GenerateMap with only the args that we want to supply.